frontend/src/scenes/surveys/SurveyEdit.tsx

```diff
@@ -73,6 +67,38 @@ export default function SurveyEdit(): JSX.Element {
         setSelectedPageIndex(newIndex)
     }
 
+    const initialThankYouMessageDescriptionContentType = surveysHTMLAvailable
+        ? thankYouMessageDescriptionContentType || 'text'
+        : 'text'
+
+    const handleSurveyValueChange = (key: string, val: string): void => {
+        const updatedAppearance = {
+            ...survey.appearance,
+            [key]: val,
+        }
+        setSurveyValue('appearance', updatedAppearance)
+    }
+
+    const updateWritingHTMLDescription = (val: boolean): void => {
+        if (surveysHTMLAvailable) {
+            handleSurveyValueChange('thankYouMessageDescriptionContentType', val ? 'html' : 'text')
+        }
+    }
+
+    const onTabChange = (key: string): void => {
+        if (surveysHTMLAvailable || key === 'text') {
+            updateWritingHTMLDescription(key === 'html')
+        }
+    }
```

Comment:
I'm not sure what the benefit is of splitting this into three methods. I think this could be a single handler, passed inline into the HTMLEditor component.